### PR TITLE
changed: put application screensaver/dpms/idle handling in separate class

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Put some members and methods from CApplication in separate class.

## Motivation and context
Split up the huge class a little.

## How has this been tested?
It builds and runs.

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/application/ApplicationPowerHandling.h [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  void ResetShutdownTimers();
+  void StopShutdownTimer();
+
+  bool ToggleDPMS(bool manual);
+
+  // Wakes up from the screensaver and / or DPMS. Returns true if woken up.
+  bool WakeUpScreenSaverAndDPMS(bool bPowerOffKeyPressed = false);
+
+protected:
```
Comment (enen92) [inline xbmc/application/ApplicationPowerHandling.h:44]: I assume the goal here is just to remove stuff from CApplication as an interim solution right?
Most of the stuff here likely belongs to GUI (or at least I see them as owned somehow by CGUIComponent)

Comment (notspiff) [reply xbmc/application/ApplicationPowerHandling.h:44]: some of it is somewhat interim yes, a lot of stuff that sits in CApplication can probably be moved to saner places. but before any such reorganization can be done we need an overview of dependencies within the huge capp class. this is imo a way to do that in a step-wise fashion, and in itself having value as things get compartmentalized (even though it's all pulled back into the monolith through inheritance). here you can now clearly see that the only thing that ties the powerhandling to the rest of the CApplication class is the appPlayer member which is passed by ref in the constructor.

Comment (enen92) [reply xbmc/application/ApplicationPowerHandling.h:44]: Yeap that's what I thought :+1: 
Thanks much for going through the pain of cleaning this gigantic piece of crap

**Comment by lrusak** [comment]
I don't have time to review this right now, but I just wanted to leave a general comment. I was working on splitting the GUI thread. The main work is in this commit https://github.com/lrusak/xbmc/commit/70b6fd2d2a46e426f38591384e894688ad66a8e3

I imagine this PR will make it easier to do some of the split.

The full branch is here https://github.com/lrusak/xbmc/commits/servicebroker-nullptr

Any help towards the direction of split gui thread and headless mode is great!

File: xbmc/application/ApplicationPowerHandling.h (hunk 1 above)
Comment (garbear) [reply xbmc/application/ApplicationPowerHandling.h:46]: Indeed. The first step is to identify dependencies by seeing how far we can separate the code.

File: xbmc/Application.h [hunk 2]
```diff
@@ -125,7 +122,8 @@ class CApplication : public IWindowManagerCallback,
                      public ISettingCallback,
                      public ISettingsHandler,
                      public ISubSettings,
-                     public KODI::MESSAGING::IMessageTarget
+                     public KODI::MESSAGING::IMessageTarget,
+                     public CApplicationPowerHandling
```
Comment (garbear) [inline xbmc/Application.h:126]: I assume you're using inheritance for composition? Sometimes there's better ways to compose components.

Comment (notspiff) [reply xbmc/Application.h:126]: yes. but i don't want to change the approach now. i just want to split the class in logical bits, then we can see about is-a vs has-a later.

**Comment by notspiff** [comment]
jenkins build this please

**Review by garbear — APPROVED** [review]

**Comment by notspiff** [comment]
Approved, all green, no clang format noise. Merging.

**Pull request merged**